src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/AttributeSnippetFormatOnTypeProvider.cs

```diff
@@ -0,0 +1,78 @@
+
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Formatting
+{
+    internal class AttributeSnippetFormatOnTypeProvider : RazorFormatOnTypeProvider
+    {
+        public override string TriggerCharacter => "=";
+
+        public override bool TryFormatOnType(Position position, FormattingContext context, out TextEdit[] edits)
+        {
+            if (position is null)
+            {
+                throw new ArgumentNullException(nameof(position));
+            }
+
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            bool addCursorPlaceholder;
+            if (context.Options.TryGetValue(LanguageServerConstants.ExpectsCursorPlaceholderKey, out var value) && value.IsBool)
+            {
+                addCursorPlaceholder = value.Bool;
+            }
+            else
+            {
+                // Temporary:
+                // no-op if cursor placeholder isn't supported. This means the request isn't coming from VS.
+                edits = null;
+                return false;
+            }
+
+            if (!IsAtAttributeValueStart(context, position))
+            {
+                edits = null;
+                return false;
+            }
+
+            // We've just typed a Razor comment start.
+            var cursorPlaceholder = addCursorPlaceholder ? LanguageServerConstants.CursorPlaceholderString : string.Empty;
+            var edit = new TextEdit()
+            {
+                NewText = $"=\"{cursorPlaceholder}\"",
+                Range = new Range(position, position)
+            };
+
+            edits = new[] { edit };
+            return true;
+        }
+
+        private static bool IsAtAttributeValueStart(FormattingContext context, Position position)
+        {
+            var syntaxTree = context.CodeDocument.GetSyntaxTree();
+
+            var absoluteIndex = position.GetAbsoluteIndex(context.SourceText);
+            var change = new SourceChange(absoluteIndex, 0, string.Empty);
+            var owner = syntaxTree.Root.LocateOwner(change);
+
+            return owner != null &&
+                owner.Kind == SyntaxKind.MarkupTagHelperAttribute &&
+                owner.Parent is MarkupTagHelperAttributeValueSyntax attributeValue &&
+                attributeValue.Span.IsEmpty &&
+                owner.Parent.Parent is MarkupTagHelperAttributeSyntax attribute &&
+                !attribute.Span.IsEmpty &&
+                context.SourceText[absoluteIndex - 1] == '=';
```

Comment:
This last check seems overkill given we already have the `=` trigger character. Any objections in removing it?

Reply:
Definitely remove it. I would however add a check to validate that the attribute is not expecting a "string" value because that would then result in HTML handling that case I believe.

Reply:
Sure 👍